File: servicedesk/libs/shared-types/src/lib/ticket.ts
export type TicketStatus = 'open' | 'in-progress' | 'resolved' | 'closed';
export type TicketPriority = 'low' | 'medium' | 'high' | 'critical';

export interface Ticket {
  id: string;
  title: string;
  description: string;
  status: TicketStatus;
  priority: TicketPriority;
  createdAt: string; // ISO 8601
  updatedAt: string; // ISO 8601
  assigneeId: string | null;
  reporterId: string;
}

export interface CreateTicketDto {
  title: string;
  description: string;
  priority: TicketPriority;
  reporterId: string;
}

export interface UpdateTicketDto {
  title?: string;
  description?: string;
  status?: TicketStatus;
  priority?: TicketPriority;
  assigneeId?: string | null;
}

File: servicedesk/apps/api/Models/Ticket.cs
namespace api.Models;

// Mirrors the Ticket interface in libs/shared-types/src/lib/ticket.ts.
// Keep both in sync when adding fields.

public enum TicketStatus
{
    Open,
    InProgress,
    Resolved,
    Closed
}

public enum TicketPriority
{
    Low,
    Medium,
    High,
    Critical
}

public class Ticket
{
    public required string Id { get; init; }
    public required string Title { get; init; }
    public required string Description { get; init; }
    public required TicketStatus Status { get; init; }
    public required TicketPriority Priority { get; init; }
    public required DateTimeOffset CreatedAt { get; init; }
    public required DateTimeOffset UpdatedAt { get; init; }
    public string? AssigneeId { get; init; }
    public required string ReporterId { get; init; }
}
